### PR TITLE
I finish task 3

### PR DESCRIPTION
**Pull request by XingWei0920**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -2,7 +2,9 @@ const app = require("../app");
 const request = require("supertest");
 const db = require("../db/connection");
 const seed = require("../db/seeds/seed");
+const endpoints= require("../endpoints.json")
 const { articleData, commentData, topicData, userData } = require("../db/data/test-data/index");
+const { expect } = require("@jest/globals");
```
Comment (ghost) [inline __tests__/app.test.js]: This shouldn't be here, watch out for random imports

File: __tests__/app.test.js [hunk 2]
```diff
@@ -46,5 +48,17 @@ describe("/api/topics", () => {
...
   });
+})
+
+describe("/api", () => {
+  test("GET:200 sends an object with all endpoints information", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then((response) => {
+        expect(typeof response.body.msg).toBe('object')
```
Comment (ghost) [inline __tests__/app.test.js]: What data type is an array?

File: controllers/server-controllers.js [hunk 3]
```diff
@@ -10,6 +10,15 @@ exports.getTopics = (req, res, next) => {
     .catch(next)
 }
 
+exports.getAllEndpoints= (req, res, next) => {
+    selectAllEndpoints()
+    .then((allEndpoints)=>{
+        console.log(JSON.parse(allEndpoints))
```
Comment (ghost) [inline controllers/server-controllers.js]: Don't leave console logs in

File: models/server-models.js [hunk 4]
```diff
@@ -7,4 +9,11 @@ exports.selectTopics = (req, res, next) => {
     return db.query(queryString).then((result) => {
         return result.rows;
       });
+}
+
+exports.selectAllEndpoints= (req, res, next) => {
```
Comment (ghost) [inline models/server-models.js]: You don't need to import in a json file, to convert it to a string, and then parse it back to json in your controller. You can just serve the json file back from the controller to the client

**Review by ghost — APPROVED** [review]
Some things to tidy up, this does not need resubmitting. Take a look at the feedback notes, make the changes then merge it and move on.

File: __tests__/app.test.js (hunk 2 above)
Comment (XingWei0920) [reply __tests__/app.test.js]: the data type of an array is an object.

**Pull request merged**